src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -2373,4 +2373,41 @@ public function testAccessToParentOptionFromNestedNormalizerAndLazyOption()
         ];
         $this->assertSame($expectedOptions, $actualOptions);
     }
+
+    public function testResolveOptionsDefinedByOptionConfigurator()
+    {
+        $this->resolver->define('foo')
+            ->required()
+            ->deprecated()
+            ->default('bar')
+            ->allowedTypes('string')
+            ->allowedValues(['bar', 'baz']);
+
+        $this->assertTrue($this->resolver->isDeprecated('foo'));
+        $this->assertTrue($this->resolver->isRequired('foo'));
+        $this->assertSame(['foo' => 'bar'], $this->resolver->resolve());
```

Comment:
You could use the `OptionsResolverIntrospector` helper to check other methods as well, e.g:
```php
        $this->resolver->define('foo')
            ->required()
            ->deprecated()
            ->default('bar')
            ->allowedTypes('string')
            ->addAllowedTypes('null')
            ->allowedValues('bar')
            ->addAllowedValues('baz')
            ->normalize(static function (Options $options, $value) {
                return $value;
            })
            ->addNormalizer(static function (Options $options, $value) {
                return $value;
            }, true)
        ;
        $introspector = new OptionsResolverIntrospector($this->resolver);

        $this->assertTrue(true, $this->resolver->isDefined('foo'));
        $this->assertTrue(true, $this->resolver->isDeprecated('foo'));
        $this->assertTrue(true, $this->resolver->hasDefault('foo'));
        $this->assertSame('bar', $introspector->getDefault('foo'));
        $this->assertSame(['string', 'null'], $introspector->getAllowedTypes('foo'));
        $this->assertSame(['bar', 'baz'], $introspector->getAllowedValues('foo'));
        $this->assertCount(2, $introspector->getNormalizers('foo'));
```